addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjection.java

```diff
@@ -0,0 +1,687 @@
+                        timeLong /= 10;
+                    }
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(
+                                "Caught "
+                                        + ex.getClass().getName()
+                                        + " "
+                                        + ex.getMessage()
+                                        + " when "
+                                        + IO_EX_LOG
+                                        + "due to a socket timeout, trying with the lowest interval("
+                                        + timeLong
+                                        + ")");
+                    }
+                } catch (IOException ex) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug(
+                                "Caught "
+                                        + ex.getClass().getName()
+                                        + " "
+                                        + ex.getMessage()
+                                        + " when "
+                                        + IO_EX_LOG);
+                    }
+                    return;
+                }
+            }
+        }
+        // json query injection
+        if (doJsonScan && isJsonPayload) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("\nStarting with the json query injection package of attack rules:");
+            }
+            for (String[] jpv : JSON_INJECTION) {
+                try {
+                    if (isStop()) {
+                        if (LOG.isDebugEnabled()) {
+                            LOG.debug(STOP_LOG);
+                        }
+                        return;
+                    }
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("\nTrying with the value: " + jpv[0]);
+                    }
+                    String valueInj = getParamJsonString(param, jpv);
+                    msgInjAttack = getNewMsg();
+                    setParameter(msgInjAttack, param, valueInj);
+                    sendAndReceive(msgInjAttack);
+                    bodyInjAttack = msgInjAttack.getResponseBody().toString();
+                    if (msgInjAttack.getResponseHeader().getStatusCode() != HttpStatusCode.OK) {
+                        continue;
+                    }
+                    if (!bodyBase.equals(bodyInjAttack)) {
+                        // Get more confidence
+                        if (getMoreConfidence) {
+                            String secondVal =
+                                    getParamJsonString(param, new String[] {JSON_TOKEN, jpv[1]});
+                            counterProofMsg = getNewMsg();
+                            counterProofMsg
+                                    .getRequestHeader()
+                                    .setHeader(HttpRequestHeader.CONTENT_TYPE, "application/json");
```

Comment:
We should not need to override the content type (it should be already that, per variant check) nor the method (which could change the meaning of the request).